apps/settings/js/panels/date_time/panel.js

```diff
@@ -152,9 +152,7 @@ define(function(require) {
       _updateUI: function dt_updateUI() {
         this._elements.timeAutoSwitch.dataset.state =
             DateTime.clockAutoEnabled ? 'auto' : 'manual';
-        this._elements.timeAutoSwitch.hidden =
-          this._elements.timezoneInfo.hidden =
-            !(DateTime.clockAutoAvailable || DateTime.timezoneAutoAvailable);
+        this._elements.timeAutoSwitch.hidden = !DateTime.clockAutoAvailable;
```

Comment:
It's worth to put some comments here to address the story about clockAutoAvailable && timezoneAutoAvailable && timAutoSwitch here.